docs/NNNN-direct-clean-path-layout.md

```diff
@@ -0,0 +1,255 @@
+# OCFL Community Extension NNNN: Direct Clean Path Layout
+
+* **Extension Name:** NNNN-direct-clean-path-layout
+* **Authors:** Jürgen Enge, Basel
+* **Minimum OCFL Version:** 1.0
+* **OCFL Community Extensions Version:** 1.0
+* **Obsoletes:** n/a
+* **Obsoleted by:** n/a
+
+## Overview
+
+This extension can be used as root storage layout extension to map OCFL object
+identifiers to path names as well as a mapping for content path names.
+This is done by replacing or removing "dangerous characters" from names.
+
+The functionality has been derived from David A. Wheeler - "Fixing Unix/Linux/POSIX Filenames" (https://www.dwheeler.com/essays/fixing-unix-linux-filenames.html)
+
+#### Usage Scenario
+
+If you want to make sure, that the internal layout of the OCFL content 
+substructure or the object id is as close as possible to the original structure, but you want to 
+make sure, that there are no possibly dangerous file- or foldernames, which 
+may oppose with durability.
+
+One Example could be complete filesystems, which come from estates, and are not
+under control of the archivists.
+
+### Caveat
+
+#### Projection
+This extension does  provide injective (one-to-one) projections only if 
+`encodeUTF == true`.  
+If you don't want to use the longish UTF-Code replacements, first check your source to make sure, that there's no chance of 
```

Comment:
I might use "verbose" here in place of "longish".

I might also try to describe the problem more explicitly here. That is to say, what are the conditions in which different names could be mapped to the same value?